src/npcmove.cpp

```diff
@@ -1160,13 +1160,7 @@ int npc::confident_gun_mode_range( const item::gun_mode &gun, int at_recoil ) co
         return 0;
     }
 
-    double deviation = get_weapon_dispersion( gun.target, false ) + at_recoil;
-    // Halve to get expected values
-    deviation /= 2;
-    // Convert from MoA back to quarter-degrees.
-    deviation /= 15;
-
-    int ret = std::min( int( confidence_mult() * 360 / deviation ), gun->gun_range( this ) );
+    int ret = gun_engagement_range( *gun, engagement::effective_min, at_recoil );
```

Comment:
Yes.
You can change the function to return engagement ranges instead, but then there should be a new engagement type: 100% guaranteed good hit at full aim (for strictly precise setting).